src/table_properties_rc.rs

```diff
@@ -0,0 +1,201 @@
+        TablePropertiesCollectionIter::new(self.handle.clone())
+    }
+
+    pub fn len(&self) -> usize {
+        unsafe { crocksdb_ffi::crocksdb_table_properties_collection_len(self.handle.ptr()) }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+}
+
+pub struct TablePropertiesCollectionIter {
+    handle: TablePropertiesCollectionIteratorHandle,
+}
+
+impl TablePropertiesCollectionIter {
+    fn new(collection: TablePropertiesCollectionHandle) -> TablePropertiesCollectionIter {
+        TablePropertiesCollectionIter {
+            handle: TablePropertiesCollectionIteratorHandle::new(collection),
+        }
+    }
+}
+
+impl Iterator for TablePropertiesCollectionIter {
+    type Item = (TablePropertiesKey, TableProperties);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        unsafe {
+            loop {
+                if !crocksdb_ffi::crocksdb_table_properties_collection_iter_valid(self.handle.ptr())
+                {
+                    return None;
+                }
+
+                let mut keylen: size_t = 0;
+                let key = crocksdb_ffi::crocksdb_table_properties_collection_iter_key(
+                    self.handle.ptr(),
+                    &mut keylen,
+                );
+                let props = crocksdb_ffi::crocksdb_table_properties_collection_iter_value(
+                    self.handle.ptr(),
+                );
+                crocksdb_ffi::crocksdb_table_properties_collection_iter_next(self.handle.ptr());
+                if !props.is_null() {
+                    assert!(!key.is_null() && keylen != 0);
+                    let key = TablePropertiesKey::new(key, keylen, self.handle.clone());
+                    let props_handle = TablePropertiesHandle::new(props, self.handle.clone());
+                    let val = TableProperties::new(props_handle);
+                    return Some((key, val));
+                }
+            }
+        }
+    }
+}
+
+/// # Safety
+///
+/// The underlying iterator is over an unordered map of heap-allocated strings,
+/// so as long as the iterator and collection are alive, the key pointers are
+/// valid.
+pub struct TablePropertiesKey {
+    key: *const u8,
+    keylen: size_t,
+    _iter_handle: TablePropertiesCollectionIteratorHandle,
+}
+
+impl TablePropertiesKey {
+    fn new(
+        key: *const u8,
+        keylen: size_t,
+        _iter_handle: TablePropertiesCollectionIteratorHandle,
+    ) -> TablePropertiesKey {
+        TablePropertiesKey {
+            key,
+            keylen,
+            _iter_handle,
+        }
+    }
+}
+
+impl Deref for TablePropertiesKey {
+    type Target = str;
+
+    fn deref(&self) -> &str {
+        // Safety: creating slice from values reported by rocksdb, that should
+        // be valid as long is this object is valid.
+        unsafe {
+            let bytes = slice::from_raw_parts(self.key, self.keylen);
+            str::from_utf8(bytes).unwrap()
```

Comment:
🤔 not sure if it's a good idea to put this O(n) utf-8 check inside `deref()`. It is fine if practically the strings are short.

Reply:
Good catch. I've put the check in the ctor.